### PR TITLE
kernel: bump 5.4 to 5.4.73

### PR DESCRIPTION
**Pull request by graysky2**
Removed upstreamed patches:
```
 generic/pending-5.4
   445-mtd-spinand-gigadevice-Only-one-dummy-byte-in-QUA.patch
   446-mtd-spinand-gigadevice-Add-QE-Bit.patch
 pistachio/patches-5.4
   150-pwm-img-Fix-null-pointer-access-in-probe.patch
```
Manually rebased:
```
 layerscape/patches-5.4
   801-audio-0011-Revert-ASoC-fsl_sai-add-of_match-data.patch
   801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch ***
   801-audio-0073-MLK-21957-3-ASoC-fsl_sai-add-bitcount-and-timestamp-.patch
   820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch
```
All modifications made by `update_kernel.sh`

```
Build system: x86_64
Build-tested: ipq806x/R7800, ath79/generic, bcm27xx/bcm2711,
              mvebu (mamba, rango), x86_64, ramips/mt7621
Run-tested: ipq806x/R7800, mvebu (mamba, rango), x86_64, ramips (RT-AC57U)
```

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

### CODE REVIEW

**Comment by graysky2** [comment]
@ViorelSuman - The rebase to patch I highlighted with *** above should be reviewed by you.

**Comment by graysky2** [comment]
@ViorelSuman -  Could you review the rebase to patch I highlighted with *** above?

File: target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch [hunk 1]
```diff
@@ -44,23 +44,23 @@ Reviewed-by: Peter Chen <peter.chen@nxp.com>
...
 --- a/drivers/usb/dwc3/core.h
 +++ b/drivers/usb/dwc3/core.h
-@@ -1041,6 +1041,8 @@ struct dwc3_scratchpad_array {
+@@ -1045,6 +1045,8 @@ struct dwc3_scratchpad_array {
   * 	2	- No de-emphasis
   * 	3	- Reserved
   * @dis_metastability_quirk: set to disable metastability quirk.
 + * @host_vbus_glitches: set to avoid vbus glitch during
 + *                      xhci reset.
+  * @dis_split_quirk: set to disable split boundary.
```
Comment (adschm) [inline target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch:53]: Consider putting dis_split_quirk before host_vbus_glitches here and in the next change, in order to keep local alphabetic order and more reasonable groups.

**Comment by adschm** [comment]
Looks good from looking at it. For this size of the bump, I think we should have some more tests ...

File: target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch:53]: @adschm - Can one do this easily using quilt?  The alternative is to untar the source and edit, then rebuild the patch.  Perhaps there is an easier method?

Comment (adschm) [reply target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch:53]: Personally, I would actually just have edited the patch manually, and then run the refresh script on the layerscape target only.

Comment (graysky2) [reply target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch:53]: Do you mind pasting the manually modified patch as a gist?  I will use it, re-run the script, and rebase/push.

File: target/linux/layerscape/patches-5.4/801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch [hunk 2]
```diff
@@ -36,7 +36,7 @@ Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
 +	return 0;
 +}
 +
- static struct snd_soc_dai_driver fsl_sai_dai = {
```
Comment (ViorelSuman) [inline target/linux/layerscape/patches-5.4/801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch:39]: Change from "fsl_sai_dai" to "fsl_sai_dai_template" is not part of the original patch, it was done later in a different context. Now it depends on the order the patches are applied - the kernel should be compilable with no errors after each patch is applied, one by one.

Comment (graysky2) [reply target/linux/layerscape/patches-5.4/801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch:39]: @ViorelSuman - I do not find a reference to "fsl_sai_dai" in the PR.... just:
* fsl_sai_dai_probe(struct snd_
* fsl_sai_dai_resume(struct snd_soc_dai *cpu_dai)
* fsl_sai_dai_template

Comment (ViorelSuman) [reply target/linux/layerscape/patches-5.4/801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch:39]: Right, ok, so you apply those patches on top of the upstream version where "fsl_sai_dai_template" is present I think.
Then it's fine with me, your patch is ok.

Comment (graysky2) [reply target/linux/layerscape/patches-5.4/801-audio-0039-MLK-16224-6-ASoC-fsl_sai-fix-DSD-suspend-resume.patch:39]: Thank you for the time to review

**Comment by anomeome** [comment]
compile / test : mvebu / mamba, rango

**Comment by curtdept** [comment]
Ill run a compile at least and hopefully a runtest and update this comment. Supposedly 5.10 is new LTS as well, might be worth preworking that.

File: target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch:53]: I saved your patch (omitting the last line containing the ...) but it did not apply cleanly.

```
% patch -Np1 -i fix.patch
patching file target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch
Hunk #1 FAILED at 32.
Hunk #2 FAILED at 44.
2 out of 2 hunks FAILED -- saving rejects to file target/linux/layerscape/patches-5.4/820-usb-0009-usb-dwc3-Add-workaround-for-host-mode-VBUS-glitch-wh.patch.rej
```

**Comment by graysky2** [comment]
Can someone give me a lesson in resolving the conflict since master changed `include/kernel-version.mk` since I submitted this?

I changed `include/kernel-version.mk` removing the two lines relating to 4.19, rebased and force pushed but there are still conflicts.

**Comment by graysky2** [comment]
Rebased and updated to resolve the conflict in `include/kernel-version.mk`

**Comment by Borromini** [comment]
Works fine here on Asus RT-AC57U (mt7621).

**Comment by graysky2** [comment]
Updated message with reports of success from folks here.

**Comment by adschm** [comment]
> Can someone give me a lesson in resolving the conflict since master changed include/kernel-version.mk since I submitted this?

Normally, you rebase, and during rebase you get warned about a conflict. In this case case, you would need to fix the conflict manually, then use "git add" on the file, and finally do "git rebase --con".

**Comment by graysky2** [comment]
@adschm - I found [this nice guide](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) I used to fix.  Note that 5.4.74 is out now.  I have been running 5.4.74-rc2 just fine so am preparing to adjust the PR for it unless you see any value in merging prior.

**Comment by adschm** [comment]
I think I will merge .73 today, so maybe just wait.

**Comment by graysky2** [comment]
@adschm - OK.  I will send another PR after you merge this one.

**Comment by adschm** [comment]
Thanks, merged.

**Pull request closed without merging**